pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -483,7 +483,54 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 				if err != nil {
 					log.Printf("PlanExecutionController: Cleanup failed: %v", err)
 				}
-				result, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, obj, func(runtime.Object) error { return nil })
+
+        arg := obj.DeepCopyObject()
+				result, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, arg, func(newObj runtime.Object) error {
+					//TODO Clean this up.  I don't like having to do a switch here
+					switch t := newObj.(type) {
+					case *appsv1.StatefulSet:
+						log.Printf("PlanExecutionController: CreateOrUpdate Type %v", t)
+
+						newSs := newObj.(*appsv1.StatefulSet)P
+						ss, ok := obj.(*appsv1.StatefulSet)
+						if !ok {
+							return fmt.Errorf("object passed in doesn't match StatefulSet type expected")
+						}
+
+						// We need some specilized logic in there.  We can't jsut copy the Spec since there are other values
+						// like spec.updateStrate, spec.volumeClaimTemplates, etc that are all
+						// generated from the object by the k8s controller.  We just want to update things we can change
+						newSs.Spec.Replicas = ss.Spec.Replicas
+
+						return nil
+
+					case *appsv1.Deployment:
+						newD := newObj.(*appsv1.Deployment)
+						d, ok := obj.(*appsv1.Deployment)
+						if !ok {
+							return fmt.Errorf("object passed in doesn't match deployment type expected")
+						}
+						newD.Spec.Replicas = d.Spec.Replicas
+						return nil
+
+					case *batchv1.Job:
+						// job := obj.(*batchv1.Job)
+
+					case *maestrov1alpha1.Instance:
+						// i := obj.(*maestrov1alpha1.Instance)
+
+					//unless we build logic for what a healthy object is, assume its healthy when created
+					default:
+						log.Printf("PlanExecutionController: Type %v CreateOrUpdate is not implemented yet\n", t)
```

Comment:
s/\n/
`%+v`

Reply:
I actually prefer
```
						log.Print("PlanExecutionController: CreateOrUpdate: Type is not implemented yet")
```

Otherwise the entire obj is being printed to `stdout` which looks really not easy on the eye:

```
2019/02/07 12:51:01 PlanExecutionController: CreateOrUpdate Type &StatefulSet{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:kafka-kafka,GenerateName:,Namespace:default,SelfLink:/apis/apps/v1/namespaces/default/statefulsets/kafka-kafka,UID:dd70e40c-2aff-11e9-b462-b67c7107056a,ResourceVersion:7461,Generation:11,CreationTimestamp:2019-02-07 09:43:23 -0800 PST,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{app: kafka,heritage: maestro,instance: kafka,kafka: kafka,phase: deploy-kafka,plan: deploy,planexecution: kafka-deploy-94924000,step: deploy,version: 2.11-2.4.0,},Annotations:map[string]string{},OwnerReferences:[{maestro.k8s.io/v1alpha1 Instance kafka dc238cfd-2aff-11e9-b462-b67c7107056a 0xc0008ec720 0xc0008ec721}],Finalizers:[],ClusterName:,Initializers:nil,},Spec:StatefulSetSpec{Replicas:*0,Selector:&k8s_io_apimachinery_pkg_apis_meta_v1.LabelSelector{MatchLabels:map[string]string{app: kafka,heritage: maestro,instance: kafka,kafka: kafka,phase: deploy-kafka,plan: deploy,planexecution: kafka-deploy-94924000,step: deploy,version: 2.11-2.4.0,},MatchExpressions:[],},Template:k8s_io_api_core_v1.PodTemplateSpec{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{app: kafka,heritage: maestro,instance: kafka,kafka: kafka,phase: deploy-kafka,plan: deploy,planexecution: kafka-deploy-94924000,step: deploy,version: 2.11-2.4.0,},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,},Spec:PodSpec{Volumes:[{config {nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ConfigMapVolumeSource{LocalObjectReference:LocalObjectReference{Name:kafka-serverproperties,},Items:[],DefaultMode:*420,Optional:nil,} nil nil nil nil nil nil nil nil}}],Containers:[{k8skafka gcr.io/google_samples/k8skafka:v1 [sh -c exec kafka-server-start.sh /config/server.properties --override broker.id=${HOSTNAME##*-}] []  [{server 0 9093 TCP }] [] [{KAFKA_HEAP_OPTS -Xmx512M -Xms512M nil} {KAFKA_OPTS -Dlogging.level=INFO nil}] {map[] map[cpu:{{200 -3} {<nil>} 200m DecimalSI} memory:{{200 -3} {<nil>} 200m DecimalSI}]} [{datadir false /var/lib/kafka  <nil>} {config false /config  <nil>}] [] nil Probe{Handler:Handler{Exec:&ExecAction{Command:[sh -c /opt/kafka/bin/kafka-broker-api-versions.sh --bootstrap-server=localhost:9093],},HTTPGet:nil,TCPSocket:nil,},InitialDelaySeconds:0,TimeoutSeconds:1,PeriodSeconds:10,SuccessThreshold:1,FailureThreshold:3,} nil /dev/termination-log File Always nil false false false}],RestartPolicy:Always,TerminationGracePeriodSeconds:*300,ActiveDeadlineSeconds:nil,DNSPolicy:ClusterFirst,NodeSelector:map[string]string{},ServiceAccountName:,DeprecatedServiceAccount:,NodeName:,HostNetwork:false,HostPID:false,HostIPC:false,SecurityContext:&PodSecurityContext{SELinuxOptions:nil,RunAsUser:*1000,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:*1000,RunAsGroup:nil,Sysctls:[],},ImagePullSecrets:[],Hostname:,Subdomain:,Affinity:nil,SchedulerName:default-scheduler,InitContainers:[],AutomountServiceAccountToken:nil,Tolerations:[],HostAliases:[],PriorityClassName:,Priority:nil,DNSConfig:nil,ShareProcessNamespace:nil,ReadinessGates:[],},},VolumeClaimTemplates:[{{ } {datadir      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[phase:deploy-kafka plan:deploy planexecution:kafka-deploy-94924000 step:deploy version:2.11-2.4.0 app:kafka heritage:maestro instance:kafka] map[] [] nil [] } {[ReadWriteOnce] nil {map[] map[storage:{{1073741824 0} {<nil>} 1Gi BinarySI}]}  <nil> <nil>} {Pending [] map[] []}}],ServiceName:kafka-svc,PodManagementPolicy:OrderedReady,UpdateStrategy:StatefulSetUpdateStrategy{Type:RollingUpdate,RollingUpdate:&RollingUpdateStatefulSetStrategy{Partition:*0,},},RevisionHistoryLimit:*10,},Status:StatefulSetStatus{ObservedGeneration:11,Replicas:0,ReadyReplicas:0,CurrentReplicas:0,UpdatedReplicas:0,CurrentRevision:kafka-kafka-76f9b66b94,UpdateRevision:kafka-kafka-76f9b66b94,CollisionCount:*0,Conditions:[],},}
```

turns into:
```
2019/02/07 12:54:17 PlanExecutionController: CreateOrUpdate: Type is not implemented yet
```